src/agent/onefuzz-telemetry/src/lib.rs

```diff
@@ -1,13 +1,36 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+use serde::{Deserialize, Serialize};
 use std::sync::{LockResult, RwLockReadGuard, RwLockWriteGuard};
 use uuid::Uuid;
 
 pub use appinsights::telemetry::SeverityLevel::{Critical, Error, Information, Verbose, Warning};
 
-pub type MicrosoftTelemetryKey = Uuid;
-pub type InstanceTelemetryKey = Uuid;
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct MicrosoftTelemetryKey(Uuid);
+impl MicrosoftTelemetryKey {
+    pub fn to_string(&self) -> String {
```

Comment:
Consider adding actual impls for [`std::string::ToString`](https://doc.rust-lang.org/std/string/trait.ToString.html), or even better, impl [`Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html), which could be useful generally, and triggers a [blanket impl](https://doc.rust-lang.org/std/string/trait.ToString.html#impl-ToString-4) for `ToString`.